### PR TITLE
feat: add `<FormLayout />` component

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds `<FormLayout />` component

## ✅ Checklist

- [x] Visuals are complete and match Figma
- [x] Code is complete and in accordance with our style guide
- [x] Design and theme tokens are audited for any relevant changes
- [x] Unit tests are written and passing
- [x] TSDoc is written or updated for any component API surface area
- [x] Stories in Storybook accompany any relevant component changes
- [x] Ensure no accessibility violations are reported in Storybook
- [x] Specs and documentation are up-to-date
- [x] Cross-browser check is performed (Chrome, Safari, Firefox)
- [x] Changeset is added

### CODE REVIEW

File: documentation/specs/FormLayout.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ type FormLayoutTitleProps = {
 };
 
 type FormLayoutHelperTextProps = {
+  id?: string;
```
Comment (stephenjwatkins) [inline documentation/specs/FormLayout.md:45]: potentially used for `aria-describedby` on parent form element

File: easy-ui-react/src/FormLayout/Section.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    </WithinSectionContext.Provider>
+  );
+}
+
+Section.displayName = "FormLayout.Section";
+
+function findSectionTitleInChildren(children: ReactNode) {
+  // look for the first `FormLayout.Title` or `FormLayout.Section` element.
+  // if we encounter a Section before a Title we know the outer Section
+  // doesn't have a title (or it's wildly [and incorrectly] out of order)
```
Comment (OskiTheCoder) [inline easy-ui-react/src/FormLayout/Section.tsx:51]: **thought**: not related to whats going on in this function, but this comment got me thinking about error handling for components in easy ui that require some ordering or to be structured in some way. i haven't seen how other libraries deal with it, if it all, but i did some minor handling in the search nav and at the time it made sense for me to do it, but i wonder what the best practices are for these kinds of things.

File: easy-ui-react/src/utilities/react.ts [hunk 3]
```diff
@@ -94,3 +94,65 @@ export function getFlattenedKey(
...
+
+/**
+ * Recursively searches a React tree finding the first that meets the specified
+ * comparator function.
+ *
+ * @param children Tree of children to search
+ * @param deepFindFn Comparator function to test each child against
+ * @returns The first child that meets the criteria of the comparator function
+ */
+export function deepFind(
```
Comment (OskiTheCoder) [inline easy-ui-react/src/utilities/react.ts:129]: **praise**: i like this utility here

File: easy-ui-react/src/FormLayout/Section.tsx (hunk 2 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/FormLayout/Section.tsx:51]: true. it's a good question. i think some libraries handle this situation with "slots", which i think i've seen comes with some error handling, but i'll do some research for future iterations

File: easy-ui-react/src/FormLayout/HelperText.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React, { ReactNode } from "react";
+import { Text } from "../Text";
+
+export type HelperTextProps = {
+  /**
+   * Id to assign to the HTML element.
+   */
+  id?: string;
```
[the remark this answers is not in the document]
Comment (stephenjwatkins) [reply easy-ui-react/src/FormLayout/HelperText.tsx:8]: lol yes. good catch

**Review by OskiTheCoder — APPROVED** [review]
awesome!

**Pull request merged**